### PR TITLE
Stage lvuidf 24

### PR DESCRIPTION
**Pull request by Fleeck**
Subscribe button + static pages : About school, about teachers, contacts.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -27,4 +27,4 @@ end
 gem 'rails_12factor', group: :production
 gem 'geocoder'
 
-ruby '2.2.2'
+ruby '2.2.2'
```
Comment (rtriska) [inline Gemfile]: add new line in the end of file

File: app/assets/javascripts/pages.coffee [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
+# All this logic will automatically be available in application.js.
+# You can use CoffeeScript in this file: http://coffeescript.org/
```
Comment (rtriska) [inline app/assets/javascripts/pages.coffee]: coffee file??? According to convention we use simple javascript

File: app/views/greet/index.html [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+		e.type = 'text/javascript'; 
+		e.async = true; 
+		e.src = ('https:' == document.location.protocol ? 'https' : 'http') + '://btn.createsend1.com/js/sb.min.js?v=3'; 
+		e.className = 'createsend-script'; 
+		var s = document.getElementsByTagName('script')[0]; 
+		s.parentNode.insertBefore(e, s); 
+	})();
+	</script>
+  </body>
+</html>
```
Comment (rtriska) [inline app/views/greet/index.html]: what is it?) just interesting :)

File: app/views/layouts/_nav.html.haml [hunk 4]
```diff
@@ -5,12 +5,12 @@
     %div
       %ul.nav.navbar-nav
         %li.dropdown
-          = link_to 'About us<span class="caret"></span>'.html_safe, '#', :class => "dropdown-toggle", 'data-toggle' => "dropdown"
+          = link_to 'About us<span class="caret"></span>'.html_safe, '#', :class => "dropdown-toggle", 'data-toggle' => "dropdown"
```
Comment (rtriska) [inline app/views/layouts/_nav.html.haml]: I think we can do this in a bit prettier way:
Please check here: http://stackoverflow.com/questions/9618971/put-haml-tags-inside-link-to-helper

File: app/views/layouts/_nav.html.haml [hunk 5]
```diff
@@ -38,6 +39,11 @@
             = link_to 'Log Out <span class="glyphicon glyphicon-log-in"></span>'.html_safe, logout_path
         - else
           %li
+            .createsend-button{"data-listid" => "i/0D/1DD/423/9C910232FABF42A1"}
+            %script(src="/assets/javascripts/subscribe_button.js")
```
Comment (rtriska) [inline app/views/layouts/_nav.html.haml]: to include js file use: http://apidock.com/rails/ActionView/Helpers/AssetTagHelper/javascript_include_tag

File: app/views/layouts/_nav.html.haml [hunk 6]
```diff
@@ -38,6 +39,11 @@
             = link_to 'Log Out <span class="glyphicon glyphicon-log-in"></span>'.html_safe, logout_path
         - else
           %li
+            .createsend-button{"data-listid" => "i/0D/1DD/423/9C910232FABF42A1"}
```
Comment (rtriska) [inline app/views/layouts/_nav.html.haml]: for data attributes we can use a bit different syntax:
{ data: { listid: <here is your id> } } and we will get the same result

File: app/views/layouts/_nav.html.haml [hunk 7]
```diff
@@ -26,7 +26,8 @@
             %li
               = link_to 'Articles', articles_path
         %li
-          = link_to "Contacts", '#'
+          = link_to 'Contacts', page_path('contacts')
+
       %ul.nav.navbar-nav.navbar-right
         - flash.each do |key, value|
           = content_tag(:li, class: "alert alert-#{key}") do
```
Comment (rtriska) [inline app/views/layouts/_nav.html.haml]: Lets follow one syntax for hashes
let it be    -    key: 'value'    not     :key => value

File: test/controllers/pages_controller_test.rb [hunk 8]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
```
Comment (rtriska) [inline test/controllers/pages_controller_test.rb]: we don't need this file because we will use Rspec

File: app/views/pages/index.html.haml [hunk 9]
```diff
@@ -1,5 +1,7 @@
+
 .container
   .jumbotron
+
     %h2 Jiraffe School is really cool!
     %img{:alt => "jiraffe", :src => "http://www.metalinjection.net/wp-content/uploads/2014/08/Giraffe-Tongue-Orchestra.jpg", :width => "500px"}/
```
Comment (rtriska) [inline app/views/pages/index.html.haml]: why we don't store images locally and don't use http://apidock.com/rails/ActionView/Helpers/AssetTagHelper/image_tag ?

File: Gemfile.lock [hunk 10]
```diff
@@ -1,186 +0,0 @@
...
-  jquery-rails
-  pg
-  rails (= 4.2.1)
-  rails_12factor
-  react-rails (~> 1.0)
-  sass-rails (~> 5.0)
-  sdoc (~> 0.4.0)
-  spring
-  uglifier (>= 1.3.0)
-  web-console (~> 2.0)
```
Comment (rtriska) [inline Gemfile.lock]: don't commit Gemfile.lock

File: app/assets/stylesheets/pages.scss [hunk 11]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the pages controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: http://sass-lang.com/
```
Comment (rtriska) [inline app/assets/stylesheets/pages.scss]: remove it. it's empty

File: app/controllers/page_controller.rb [hunk 12]
```diff
@@ -1,4 +0,0 @@
-class PageController < ApplicationController
-  def index
-  end
-end
```
Comment (rtriska) [inline app/controllers/page_controller.rb]: remove it it's empty

File: app/controllers/pages_controller.rb [hunk 13]
```diff
@@ -0,0 +1,2 @@
+class PagesController < HighVoltage::PagesController
+end
```
Comment (rtriska) [inline app/controllers/pages_controller.rb]: Do we realy need empty controllers?

File: app/helpers/pages_helper.rb [hunk 14]
```diff
@@ -0,0 +1,2 @@
+module PagesHelper
+end
```
Comment (rtriska) [inline app/helpers/pages_helper.rb]: Do we need this empty modulE?

File: app/views/pages/about_school.html.haml [hunk 15]
```diff
@@ -0,0 +1,2 @@
+%h2 About Our School
+%p This is awesome, to be here!
```
Comment (rtriska) [inline app/views/pages/about_school.html.haml]: All views will be reworked with React

File: config/initializers/high_voltage.rb [hunk 16]
```diff
@@ -0,0 +1,4 @@
+# config/initializers/high_voltage.rb
+HighVoltage.configure do |config|
+  config.route_drawer = HighVoltage::RouteDrawers::Root
+end
```
Comment (rtriska) [inline config/initializers/high_voltage.rb]: add new line in the end of file

File: config/routes.rb [hunk 17]
```diff
@@ -1,16 +1,22 @@
...
   resources :articles
   resources :events
-
-  root 'page#index'
+  
+  root 'pages#index'
+  
 
   # The priority is based upon order of creation: first created -> highest priority.
   # See how all your routes lay out with "rake routes".
```
Comment (rtriska) [inline config/routes.rb]: React router will be responsible for routes

**Comment by rtriska** [comment]
There are a lot of files related to React. Do we include it as separated js library or as a gem?

File: app/controllers/pages_controller.rb (hunk 13 above)
Comment (Fleeck) [inline app/controllers/pages_controller.rb]: This is a controller, inherited from HighVoltage controller (gem to create static pages) . It's not working without this. Other empty files have been deleted.

**Pull request closed without merging**